File: JH/programmers/level1/약수의개수와덧셈.js
function solution(left, right) {
  let result = 0;
  for (let i = left; i <= right; i++) {
    let count = 0;
    for (let j = 1; j <= i; j++) {
      if (i % j === 0) {
        count += 1;
      }
    }

    if (count % 2 === 0) {
      result += i;
    } else if(count%2===1){
      result -= i;
    }
  }
  return result;
}

//left부터 right까지 i가 레프트와 라이트 사이의 수이고 j가 그보다 작은 값
// let count = 0
// for(let i= left; i<=right; i++){
//     for(let j = 0; j<= i; j++ ){
//         if(i>j&&i%j===0){
//             count += 1
//         }
//              j += 1
//     }
//     return i
// }

//약수의 개수를 구하는 방법
// let index = 1
// let count = 0
// while(num>=index){
//     if(num%index===0){
//     count +=1
// }
//      index +=1
// }

File: jisoo/programmers/level1/약수의_개수와_덧셈.py
"""
https://school.programmers.co.kr/learn/courses/30/lessons/77884
"""


def counting(i):
    count = 0
    for x in range(1, i + 1): #절반 숫자 넘어가는 수는 어차피 안되서 확인할 필요없음
        if i % x == 0:
            count += 1
    return count


def solution(left, right):
    answer = 0

    for i in range(left, right + 1):
        count_ = counting(i)
        if count_ % 2 == 0:
            answer += i
        else:
            answer -= i

    return answer

# 단축풀이
def solution2(left, right):
    answer = 0
    for i in range(left,right+1):
        if int(i**0.5)==i**0.5:
            answer -= i
        else:
            answer += i
    return answer
